### ISSUE TITLE
Welcome to GitHub!

### ISSUE THREAD

**Issue by ebuhle** [erlwagne/RhinoWork#1]
@erlwagne, you've officially graduated from toddler to toddler-with-a-machine-gun.

**Comment by ebuhle** [comment]
Now [add @ebuhle as a collaborator](https://stackoverflow.com/questions/7920320/adding-a-collaborator-to-my-free-github-account).

**Comment by ebuhle** [comment]
Excellent!

Now, you'll have to create a new RStudio project since you've decided on the more serious `RhinoWork` and you can't rename it. When you do so, select Use Version Control and link it to this repo. Then, once you've moved all the files into the new project, delete the old one. They should show up in the Git pane, and then you'll commit and push them to GitHub.

**Comment by erlwagne** [comment]
I already did! Create a new RStudio project, at least. I moved everything over to it, etc. But there is no Git pane, so now I'm nervous. Must be time for a new project...

**Comment by ebuhle** [comment]
Did you restart RStudio after configuring git? Did you create the new project with version control, and did it populate the project name and everything?

**Comment by erlwagne** [comment]
Yeah. But now when I do Sys.which("git") it still goes to the old git sit (usr/bin/git) instead of the one I thought I had pointed it to (usr/local/bin/git).

**Comment by ebuhle** [comment]
But it seems to be working, because I just made a project linked to it and pulled successfully. I see files.

**Comment by erlwagne** [comment]
That's probably because I uploaded them manually on Github.

**Comment by ebuhle** [comment]
DOH.

OK, but I thought you told me usr/local/bin is some kind of mirror of usr/bin?

**Comment by erlwagne** [comment]
Well, that's the one the system goes to when I ask in the shell. I'm not sure how to change the path in the shell.

**Comment by erlwagne** [comment]
Now reading Jenny Bryan Ch. 9...

**Comment by ebuhle** [comment]
Change the path in the shell? I thought you were trying to change the path in RStudio's preferences.

**Comment by erlwagne** [comment]
Sorry, yeah, that's what I mean.

**Comment by ebuhle** [comment]
And what is the error you're getting now?

**Comment by erlwagne** [comment]
No error. I can create a project if I want to.

**Comment by erlwagne** [comment]
It's just that there's no Git tab in the Files / Plots / Packages / Help / Viewer pane. Should there be? Or am I fretting for nothing? (#ToddlersWithMachineGuns)

**Comment by ebuhle** [comment]
...one linked to Git? And then can you pull from the repo?

**Comment by ebuhle** [comment]
Ahh, sorry, I see. Yes, that should be there.

**Comment by ebuhle** [comment]
I seem to recall seeing this issue in a thread earlier, either on StackOverflow or on Jenny Bryan's GitHub...

**Comment by erlwagne** [comment]
Okay. Hmm. The thing is that my shell and RStudio –> Global Options / Git/SVN all agree that Git is in usr/local/bin/git, but when I ask Sys.which in the console it still says the old spot.

**Comment by ebuhle** [comment]
This is the [thread](https://github.com/jennybc/happy-git-with-r/issues/8) I was thinking of.

**Comment by erlwagne** [comment]
Okay, I'll check it out.

**Comment by ebuhle** [comment]
Did you do [this](https://github.com/jennybc/happy-git-with-r/issues/8#issuecomment-226393064)?

**Comment by erlwagne** [comment]
Yeah.

**Comment by ebuhle** [comment]
And perhaps more relevant, [this](https://github.com/jennybc/happy-git-with-r/issues/8#issuecomment-226389757).

**Comment by erlwagne** [comment]
How does one "move it out of the way"?

**Comment by ebuhle** [comment]
Was going to ask you that. I'm not sure what that means, but there's more [here](https://github.com/jennybc/happy-git-with-r/issues/8#issuecomment-226547506). (I'm still reading the thread, hence these updates.)

**Comment by ebuhle** [comment]
The fact that you're able to see the option to create a project using Git is different from the situation Bryan's students had, though, in which Git functionality was entirely missing.

**Comment by erlwagne** [comment]
Yes. I'm a bit leery of my fix (directing it to the usr/local/bin/git), which seems to have tricked the UI, but not R itself.

**Comment by ebuhle** [comment]
Oh, I think "move usr/local/bin out of the way" is referring to the $PATH. Because a program (in this case RStudio) will search the path in order, if there is a git executable in both usr/bin and usr/local/bin but the latter comes first in the $PATH, it will find that one first.

**Comment by erlwagne** [comment]
So it finds the shadow/mirror Git but not the Git itself?

**Comment by ebuhle** [comment]
That's the suggestion. See the posts near that one for more explanation, but the test would be to edit the $PATH to put `usr/bin` (or even `usr/bin/git`) first.

However, your description of having tricked the UI but not R itself is very perceptive and quite similar to Bryan's [analysis](https://github.com/jennybc/happy-git-with-r/issues/8#issuecomment-226814424). Still haven't gotten to any resolution, but this afternoon we did see that section in her book about installing git via OS X Developer Tools instead of from the web.

**Comment by ebuhle** [comment]
Warmer...[warmer](https://github.com/jennybc/happy-git-with-r/issues/8#issuecomment-257991250)...

**Comment by ebuhle** [comment]
Have you tried RStudio menu -> View -> Show Git?

**Comment by erlwagne** [comment]
And there it is...

**Comment by ebuhle** [comment]
TA DA!!!

**Comment by erlwagne** [comment]
Doesn't show anything in the Git, though.

**Comment by ebuhle** [comment]
It won't, until you edit some files it is tracking. It shows the changes since the last commit.

**Comment by ebuhle** [comment]
But you haven't yet pulled anything,  right? So start by clicking Pull.

**Comment by erlwagne** [comment]
Ah...

**Comment by ebuhle** [comment]
And? You see your project directory magically populated with files from the repo?

**Comment by erlwagne** [comment]
It won't let me click Pull!

**Comment by ebuhle** [comment]
What do you mean, won't let you click?

**Comment by erlwagne** [comment]
Like, the Pull arrow is inactive / not highlighted / whatever.

**Comment by erlwagne** [comment]
I think I need to figure out a way to change the git in the shell.

**Comment by ebuhle** [comment]
Read through that thread, esp. Jenny Bryan's suggestions re installing git from OSX Tools. But maybe first try (1) pointing RStudio's git chooser to the path given in the last post I linked; and if that fails, (2) editing the $PATH.

**Comment by erlwagne** [comment]
Okey-dokes. We'll see what happens. And now El is coming up so I should wrap this up. Until tomorrow...

**Comment by erlwagne** [comment]
Boom, there it is, motherfucker!

**Comment by erlwagne** [comment]
Git tab is now populated with RhinoWork. I think we're in business. This feels like a good day's work. Sadly.

**Comment by erlwagne** [comment]
El just came in, looked at all of this, and said, "Buhle's teaching you how to fish!"

**Comment by ebuhle** [comment]
I'm just delighted you've managed not to hook yourself in the ear. Yet. 

Also delighted I decided to check back in one last time after seeing @mdscheuerell commented on an issue I created today in `salmonIPM`. 

This _is_ a good day's work, are you kidding me?
